go/libraries/doltcore/sqle/dolt_index.go

```diff
@@ -31,73 +32,84 @@ type DoltIndex interface {
-	if len(di.cols) != len(keys) {
-		return nil, errors.New("keys must specify all columns for index")
+	tpl, err := di.keysToTuple(keys, false)
+	if err != nil {
+		return nil, err
 	}
-	return di.keysToIter(di.keysToExpr(keys, func(left sql.Expression, right sql.Expression) sql.Expression {
-		return expression.NewGreaterThanOrEqual(left, right)
-	}))
+	readRange := &noms.ReadRange{Start: tpl, Inclusive: true, Reverse: false, Check: func(tuple types.Tuple) (bool, error) {
+		return true, nil
+	}}
+	return di.rangeToIter(readRange)
 }
 
 func (di *doltIndex) AscendLessThan(keys ...interface{}) (sql.IndexLookup, error) {
-	if len(di.cols) != len(keys) {
-		return nil, errors.New("keys must specify all columns for index")
+	tpl, err := di.keysToTuple(keys, false)
+	if err != nil {
+		return nil, err
 	}
-	return di.keysToIter(di.keysToExpr(keys, func(left sql.Expression, right sql.Expression) sql.Expression {
-		return expression.NewLessThan(left, right)
-	}))
+	readRange := &noms.ReadRange{Start: tpl, Inclusive: false, Reverse: true, Check: func(tuple types.Tuple) (bool, error) {
+		return true, nil
+	}}
+	return di.rangeToIter(readRange)
 }
 
 // TODO: rename this from AscendRange to BetweenRange or something
 func (di *doltIndex) AscendRange(greaterOrEqual, lessThanOrEqual []interface{}) (sql.IndexLookup, error) {
-	if len(di.cols) != len(greaterOrEqual) || len(di.cols) != len(lessThanOrEqual) {
-		return nil, errors.New("keys must specify all columns for index")
+	greaterTpl, err := di.keysToTuple(greaterOrEqual, false)
+	if err != nil {
+		return nil, err
+	}
+	lessTpl, err := di.keysToTuple(lessThanOrEqual, true)
+	if err != nil {
+		return nil, err
 	}
-	greaterEqualExprs := di.keysToExpr(greaterOrEqual, func(left sql.Expression, right sql.Expression) sql.Expression {
-		return expression.NewGreaterThanOrEqual(left, right)
-	})
-	lessEqualExprs := di.keysToExpr(lessThanOrEqual, func(left sql.Expression, right sql.Expression) sql.Expression {
-		return expression.NewLessThanOrEqual(left, right)
-	})
-	return di.keysToIter(expression.NewAnd(greaterEqualExprs, lessEqualExprs))
+	nbf := di.indexRowData.Format()
+	readRange := &noms.ReadRange{Start: greaterTpl, Inclusive: true, Reverse: false, Check: func(tuple types.Tuple) (bool, error) {
+		return tuple.Less(nbf, lessTpl)
+	}}
+	return di.rangeToIter(readRange)
 }
 
 func (di *doltIndex) DescendGreater(keys ...interface{}) (sql.IndexLookup, error) {
-	if len(di.cols) != len(keys) {
-		return nil, errors.New("keys must specify all columns for index")
+	tpl, err := di.keysToTuple(keys, true)
+	if err != nil {
+		return nil, err
 	}
-	return di.keysToIter(di.keysToExpr(keys, func(left sql.Expression, right sql.Expression) sql.Expression {
-		return expression.NewGreaterThan(left, right)
-	}))
+	readRange := &noms.ReadRange{Start: tpl, Inclusive: true, Reverse: false, Check: func(tuple types.Tuple) (bool, error) {
```

Comment:
This is already tested for by the many tests in `dolt_index_test.go` and even the bats tests will catch it.

Brian said it best on another comment here in the PR:
> So appendMaxValue looks like it's only true for:
> 
> DescendGreater - Should be good because it skips the current key and points to position before next possible key. Will work for partial and full keys.
> DescendLessOrEquald - Should be good because it is going past the last key and points to position before next possible key. Will work for partial and full keys.
> AscendRange - Used for the end of the range which should be good assuming range is inclusive. In this case it is the same as above.